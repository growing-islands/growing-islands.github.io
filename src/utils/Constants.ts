type BreakpointsType = {
  PC: number
  SP: number
}

export const BREAKPOINTS: BreakpointsType = {
  PC: 769,
  SP: 768,
}

export const PAGE_INFO = {
  URL: process.env.NEXT_PUBLIC_SITE_URL || 'https://',
  TWITTER: 'https://twitter.com/yumelab1214',
  FACEBOOK: 'https://www.facebook.com/yumelab.co.jp/',
}

export const INTERSECTION_OPTIONS = {
  root: null,
  rootMargin: '0px 0px -100%',
  threshold: 0,
}
