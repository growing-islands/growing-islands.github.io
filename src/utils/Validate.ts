/**
 * Form Validate関連の変数
 */

export const ERROR_MESSAGES = {
  required: 'この項目は入力必須です',
  maxLength: '入力できる文字数を超えています',
  mixLength: '文字数が足りません',
  max: '数値が大きすぎます',
  mix: '数値が小さすぎます',
  num: '半角数字で入力して下さい',
  hiragana: 'ひらがなで入力してください',
  fullKana: '全角カタカナで入力してください',
  tel: '電話番号が正しい形式ではありません',
  email: 'メールアドレスの形式が正しくありません',
  password:
    '英文字（大文字、小文字）、数字をそれぞれ最低一文字以上入れた8〜16文字で入力してください',
  dependent_char: '使用できない文字列が含まれています',
  url: '「http://」もしくは「https://」から始まるURLを入力してください',
}

// NOTE： 正規表現を記述するためにESLintを無効
/*eslint-disable no-useless-escape */
/*eslint-disable no-irregular-whitespace */
export const REGEX = {
  num: /^[0-9]+$/,
  alphanumeric: /^[0-9a-zA-Z]+$/,
  hiragana: /^[ぁ-んー　]*$/,
  fullKana: /^[ァ-ヶー　]*$/,
  halfKana: /^[ｦ-ﾟ]*$/,
  tel: /^(0{1}\d{9,10})$/,
  email:
    /^([\w!#$%&'*+\-\/=?^`{|}~]+(\.[\w!#$%&'*+\-\/=?^`{|}~]+)*|"([\w!#$%&'*+\-\/=?^`{|}~. ()<>\[\]:;@,]|\\[\\"])+")@(([a-zA-Z\d\-]+\.)+[a-zA-Z]+|\[(\d{1,3}(\.\d{1,3}){3}|IPv6:[\da-fA-F]{0,4}(:[\da-fA-F]{0,4}){1,5}(:\d{1,3}(\.\d{1,3}){3}|(:[\da-fA-F]{0,4}){0,2}))\])$/,
  password:
    /^(?=.*?[a-z])(?=.*?[A-Z])(?=.*?\d)[a-zA-Z0-9!"#$%&'()\*\+\-\.,\/:;<=>?@\[\\\]^_`{|}~]{8,16}$/,
  dependent_char:
    /^(?!.*(\u2460|\u2461|\u2462|\u2463|\u2464|\u2465|\u2466|\u2467|\u2468|\u2469|\u246a|\u246b|\u246c|\u246d|\u246f|\u2470|\u2471|\u2472|\u2473|\u2160|\u2161|\u2162|\u2163|\u2164|\u2165|\u2166|\u2167|\u2168|\u2169|\u3349|\u3314|\u3322|\u334d|\u3318|\u3327|\u3303|\u3336|\u3351|\u3357|\u330d|\u3326|\u3323|\u332b|\u334a|\u333b|\u339c|\u339d|\u339e|\u338e|\u338f|\u33c4|\u33a1|\u337b|\u301d|\u301f|\u2116|\u33cd|\u2121|\u32a4|\u32a5|\u32a6|\u32a7|\u32a8|\u3231|\u3232|\u3239|\u337e|\u337d|\u337c|\u222e|\u221f|\u22bf|\u2170|\u2171|\u2172|\u2173|\u2174|\u2175|\u2176|\u2177|\u2178|\u2179|\uffe2|\uffe4|[\uD800-\uDBFF][\uDC00-\uDFFF])).*$/,
  url: /^(https?)(:\/\/[-_.!~*\'()a-zA-Z0-9;\/?:\@&=+\$,%#]+)$/,
}
/*eslint-enable no-useless-escape */
/*eslint-enable no-irregular-whitespace */
