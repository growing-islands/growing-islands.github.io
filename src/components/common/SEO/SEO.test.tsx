import { render } from '~tests/libs/utils'
import { screen } from '@testing-library/react'
import { SEO } from '.'
import type { Props } from '.'

const metaProps: Required<Props> = {
  title: 'test title',
  description: 'test description',
  url: 'https://test.com',
  image: 'https://crogic.jp/OGP.png',
  meta: {
    ogType: 'website',
    ogSiteName: 'test siteName',
    twitterCard: 'summary_large_image',
    twitterSite: '@test_account',
  },
}

describe('components/common/SEO', () => {
  it('Snapshot', () => {
    const { asFragment } = render(<SEO title={metaProps.title} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('各種Props指定', () => {
    render(
      <SEO {...{ ...metaProps }}>
        <meta name="theme-color" content="#000" data-testid="themeColor" />
      </SEO>
    )

    // title
    expect(screen.getByTestId('title')).toHaveTextContent(metaProps.title)
    expect(screen.getByTestId('ogTitle').getAttribute('content')).toEqual(
      metaProps.title + '｜株式会社ユメラボ'
    )

    // description
    expect(screen.getByTestId('description').getAttribute('content')).toEqual(
      metaProps.description
    )
    expect(screen.getByTestId('ogDescription').getAttribute('content')).toEqual(
      metaProps.description
    )

    // URL
    expect(screen.getByTestId('ogUrl').getAttribute('content')).toEqual(
      metaProps.url
    )

    // 画像
    expect(screen.getByTestId('ogImage').getAttribute('content')).toEqual(
      metaProps.image
    )

    // その他meta
    expect(screen.getByTestId('ogType').getAttribute('content')).toEqual(
      metaProps.meta.ogType
    )
    expect(screen.getByTestId('ogSiteName').getAttribute('content')).toEqual(
      metaProps.meta.ogSiteName
    )
    expect(screen.getByTestId('twitterCard').getAttribute('content')).toEqual(
      metaProps.meta.twitterCard
    )
    expect(screen.getByTestId('twitterSite').getAttribute('content')).toEqual(
      metaProps.meta.twitterSite
    )

    // props以外のmetaタグ
    expect(screen.getByTestId('themeColor').getAttribute('content')).toEqual(
      '#000'
    )
    expect(screen.getByTestId('themeColor').getAttribute('name')).toEqual(
      'theme-color'
    )
  })
})
