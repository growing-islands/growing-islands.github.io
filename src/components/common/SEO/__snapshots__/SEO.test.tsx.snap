// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/SEO Snapshot 1`] = `
<DocumentFragment>
  <title
    data-testid="title"
  >
    test title｜株式会社ユメラボ
  </title>
  <meta
    content="initial-scale=1.0, width=device-width"
    name="viewport"
  />
  <link
    href="/favicon/favicon.ico"
    rel="shortcut icon"
  />
  <link
    href="/favicon/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="/favicon/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="/favicon/apple-touch-icon.png"
    rel="apple-touch-icon"
    type="image/png"
  />
  <link
    href="/favicon/android-chrome-192x192.png"
    rel="icon"
    sizes="192x192"
    type="image/png"
  />
  <link
    href="/favicon/android-chrome-512x512.png"
    rel="icon"
    sizes="512x512"
    type="image/png"
  />
  <meta
    content="/favicon/mstile-150×150.png"
    name="msapplication-TileImage"
  />
  <meta
    content="#1e2888"
    name="theme-color"
  />
  <meta
    content="株式会社ユメラボは千葉県浦安市のIT/Web開発・SES事業でお客様のIT投資を最大限に生かし価値を共に創り続けるシステム開発企業です。システム開発・保守運用、IT活用・サポート、SESでの体制構築から、Webサイト制作・企画運用まで、IT/Webに関わるトータルソリューションサービスを提供しております。"
    data-testid="description"
    name="description"
  />
  <meta
    content="test title｜株式会社ユメラボ"
    data-testid="ogTitle"
    property="og:title"
  />
  <meta
    content="株式会社ユメラボは千葉県浦安市のIT/Web開発・SES事業でお客様のIT投資を最大限に生かし価値を共に創り続けるシステム開発企業です。システム開発・保守運用、IT活用・サポート、SESでの体制構築から、Webサイト制作・企画運用まで、IT/Webに関わるトータルソリューションサービスを提供しております。"
    data-testid="ogDescription"
    property="og:description"
  />
  <meta
    content="website"
    data-testid="ogType"
    property="og:type"
  />
  <meta
    content=""
    data-testid="ogUrl"
    property="og:url"
  />
  <meta
    content="/OGP.png"
    data-testid="ogImage"
    property="og:image"
  />
  <meta
    content="test title｜株式会社ユメラボ"
    data-testid="ogSiteName"
    property="og:site_name"
  />
  <meta
    content="summary_large_image"
    data-testid="twitterCard"
    name="twitter:card"
  />
  <meta
    content="@yumelab1214"
    data-testid="twitterSite"
    name="twitter:site"
  />
</DocumentFragment>
`;
